test/e2e/storage/csimock/mutable_csinode_allocatable.go

```diff
@@ -0,0 +1,147 @@
+		nodeName   string
+		driverName string
+	)
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		var calls int32
+		hook := drivers.Hooks{
+			Post: func(_ context.Context, method string, _ interface{}, reply interface{}, err error) (interface{}, error) {
+				if strings.Contains(method, "NodeGetInfo") {
+					if r, ok := reply.(*csipbv1.NodeGetInfoResponse); ok && err == nil {
+						if atomic.AddInt32(&calls, 1) == 1 {
+							r.MaxVolumesPerNode = initialMaxVolumesPerNode
+						} else {
+							r.MaxVolumesPerNode = updatedMaxVolumesPerNode
+						}
+						framework.Logf("NodeGetInfo called, setting MaxVolumesPerNode to %d", r.MaxVolumesPerNode)
+						return r, nil
+					}
+				}
+				return reply, err
+			},
+		}
+
+		opts := drivers.CSIMockDriverOpts{
+			Embedded:       true,
+			RegisterDriver: true,
+			Hooks:          hook,
+		}
+		driver = drivers.InitMockCSIDriver(opts)
+		cfg = driver.PrepareTest(ctx, f)
+
+		k8s = f.ClientSet
+		driverName = cfg.GetUniqueDriverName()
+		nodeName = cfg.ClientNodeSelection.Name
+
+		enablePeriodicUpdates(ctx, k8s, driverName, updatePeriodSeconds)
+
+		err := drivers.WaitForCSIDriverRegistrationOnNode(ctx, nodeName, driverName, k8s)
+		framework.ExpectNoError(err)
+	})
+
+	f.It("should observe dynamic changes in CSINode allocatable count",
+		func(ctx context.Context) {
+			framework.Logf("Testing dynamic changes in CSINode allocatable count")
+			initVal, err := readLimit(ctx, k8s, nodeName, driverName)
+			framework.ExpectNoError(err)
+			framework.Logf("Initial MaxVolumesPerNode limit: %d", initVal)
+
+			err = wait.PollUntilContextTimeout(ctx, time.Duration(updatePeriodSeconds), timeout, true,
+				func(ctx context.Context) (bool, error) {
+					cur, err := readLimit(ctx, k8s, nodeName, driverName)
+					if err != nil {
+						return false, nil
+					}
+					return int64(cur) == updatedMaxVolumesPerNode, nil
+				})
+
+			framework.ExpectNoError(err, "CSINode allocatable count was not updated to %d in time", updatedMaxVolumesPerNode)
+			framework.Logf("SUCCESS: MaxVolumesPerNode updated limit %d", updatedMaxVolumesPerNode)
+		})
+})
+
+func enablePeriodicUpdates(ctx context.Context, cs clientset.Interface, name string, period int64) {
+	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, timeout, true,
+		func(ctx context.Context) (bool, error) {
+			obj, err := cs.StorageV1().CSIDrivers().Get(ctx, name, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			if obj.Spec.NodeAllocatableUpdatePeriodSeconds != nil && *obj.Spec.NodeAllocatableUpdatePeriodSeconds == period {
+				return true, nil
+			}
+			obj.Spec.NodeAllocatableUpdatePeriodSeconds = &period
+			_, err = cs.StorageV1().CSIDrivers().Update(ctx, obj, metav1.UpdateOptions{})
+			return err == nil, nil
+		})
+	framework.ExpectNoError(err, "enabling periodic CSINode allocatable updates failed")
+}
+
+func readLimit(ctx context.Context, cs clientset.Interface, node, drv string) (int32, error) {
```

Comment:
So these functions are will be visible to other mock tests and hence so we should have little more context in their name.  I would rename `readLimit` --> `readCSINodeLimit`. Rename `enablePeriodicUpdates` --> `updateCSINodeWithPeriodicUpdate` or `enablePeriodicUpdateOfCSINode`